error.go

```diff
@@ -1,6 +1,30 @@
 package huma
 
-// ErrorModel defines a basic error message
+import "fmt"
+
+// ErrorDetailer returns error details for responses & debugging.
+type ErrorDetailer interface {
+	ErrorDetail() *ErrorDetail
+}
+
+// ErrorDetail provides details about a specific error.
+type ErrorDetail struct {
+	Message  string      `json:"message,omitempty"`
+	Location string      `json:"location,omitempty"`
```

Comment:
What's the purpose of Location? Specifically, is it a source code location? Because an end-user won't be able to do anything with that information (other than providing it in a support ticket, in which case some sort of error code would probably be better anyway).

Reply:
I've added some additional documentation. This is actually the location in the input as a JSONPath type link and can refer to request parameters (path/query/header) or the request body. For example:

```json
{
  "status": 400,
  "title": "Bad Request",
  "detail": "Request input did not pass validation",
  "errors": [
    {
      "message": "Could not parse int",
      "location": "path.thing-id",
      "value": "abc"
    },
    {
      "message": "Field is required",
      "location": "body.tags[1].description"
    },
    {
      "message": "Must match pattern ^[a-z0-9-]+$",
      "location": "body.slug",
      "value": "foo_bar"
    }
  ]
}
```